pyriemann/estimation.py

```diff
@@ -423,40 +422,82 @@ class Coherences(CospCovariances):
         The maximal frequency to be returned.
     fs : float | None, (default None)
         The sampling frequency of the signal.
+    coh : {'ordinary', 'instantaneous', 'lagged', 'imaginary'}, (default
+            'ordinary')
+        The coherence type:
+
+        * 'ordinary' for the ordinary coherence, defined in Eq.(22) of [1]_;
+          this normalization of cross-spectral matrices captures both in-phase
+          and out-of-phase correlations. However it is inflated by the
+          artificial in-phase (zero-lag) correlation engendered by volume
+          conduction.
+        * 'instantaneous' for the instantaneous coherence, Eq.(26) of [1]_,
+          capturing only in-phase correlation.
+        * 'lagged' for the lagged-coherence, Eq.(28) of [1]_, capturing only
+          out-of-phase correlation.
+        * 'imaginary' for the imaginary coherence, Eq.(0.16) of [2]_, capturing
+          out-of-phase correlation but still affected by in-phase correlation.
+
+    Attributes
+    ----------
+    freqs_ : ndarray, shape (n_freqs,)
+        If transformed, the frequencies associated to cospectra.
+        None if ``fs`` is None.
 
     See Also
     --------
     Covariances
     HankelCovariances
     CospCovariances
+
+    References
+    ----------
+    .. [1] Pascual-Marqui R. "Instantaneous and lagged measurements of linear
+        and nonlinear dependence between groups of multivariate time series:
+        frequency decomposition", arXiv, 2007
+
+    .. [2] Congedo, M. "Non-Parametric Synchronization Measures used in EEG
+        and MEG", TechReport, 2018
```

Comment:
to me https://pubmed.ncbi.nlm.nih.gov/15351371/ is another standard reference

Reply:
also can you add the URL to pdf. Will make it easier to find equations

Reply:
As explained in Pascual Marqui 2007:

> While the instantaneous coherence is the real part of the complex valued coherency, the lagged coherence is not the imaginary part of the complex valued coherency. Ideally, the lagged coherence is a measure that *is not* affected by instantaneous dependence, whereas the imaginary part of the complex valued coherency (Nolte et al 2004) is more affected by instantaneous dependence (Pascual-Marqui 2007a).

So, with hindsight, imaginary coherence of Nolte seems an intermediary step towards the lagged-coherence. So, I am not sure that it is a standard reference.

Reply:
for me the Nolte paper is the seminal paper to have proposed the imaginary part of coherence. So at least I would point to Nolte et al for this option.

Reply:
ok, fine.